### ISSUE TITLE
Bug writing to real PostGres instance

### ISSUE THREAD

**Issue by IanMayo** [debrief/pepys-import#159]
I've created some remote PostGIS hosting.  I'll share connection details separately.

Anyway, it was failing. I tried to debug, then I realised I had the same problem locally - see below.

Note: we don't get the issue when we're working with the `testing_posgres` library in unit tests.

```
E-trac parser working on e_trac_bad.txt
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/ian/git/pepys-import/pepys_import/import.py", line 42, in <module>
    main(path=args.path, db=args.db)
  File "/Users/ian/git/pepys-import/pepys_import/import.py", line 29, in main
    processor.process(path, data_store, True)
  File "/Users/ian/git/pepys-import/pepys_import/file/file_processor.py", line 116, in process
    file, current_path, data_store, processed_ctr
  File "/Users/ian/git/pepys-import/pepys_import/file/file_processor.py", line 190, in process_file
    importer.load_this_file(data_store, full_path, file_contents, datafile)
  File "/Users/ian/git/pepys-import/pepys_import/file/e_trac_importer.py", line 43, in load_this_file
    datafile.measurements[self.short_name] = list()
AttributeError: 'Datafile' object has no attribute 'measurements'
```

**Comment by robintw** [comment]
I haven't got time to go into this in depth now, as I'm not supposed to be working - but I saw the email and it niggled at me, so I wanted to get to the bottom of it.

Basically, the `Datafile` class in `sqlite_db.py` has been updated to have `measurements` dict as an attribute, plus various changes to methods so they work with the measurements dict. The `Datafile` class in `postgres_db.py` hasn't been updated, and still has a `_measurements` list and none of the methods to work on the new `measurements` attribute.

I don't think I should try and update this, as I'm not the expert in the changes that were done to introduce the new attributes - @BarisSari did the changes for SQLite so should probably take the lead on this.

We didn't notice this earlier because - I _think_ - all of the tests that import things do so with an in-memory SQLite DB and not with Postgres. We should rectify this! In fact, possibly we should import all the sample data (basically running the import CLI on the sample data folder) for both SQLite and Postgres as a test, and check there aren't any exceptions raised.

**Comment by IanMayo** [comment]
Thanks for that Robin, I'm just coming to the same conclusion, too :-)

I'd spotted that the `measurements` variable needed to be renamed. Now I also recognise it needs to be a dictionary, and not a list.  Thanks for the pointer!

(I'd quite like us to pull this common functionality up a layer - we're duplicating it across SQLite/PostGres)

**Comment by BarisSari** [comment]
Hello everyone,

As @robintw mentioned above, `postgres_db.py` hasn't been updated. Therefore, it failed. I was careful about updating it for everything we do for SQLite, however, I missed this time. I think we should somehow find a way to prevent duplication and have common functions. I can fix it for now by duplicating codes but I suggest opening a new issue for pulling this functionality up a layer.

**Comment by IanMayo** [comment]
:+1 for refactoring to remove duplication.  Temporary fix implemented at #160
